### PR TITLE
Feature lwdev 4905 fee in operations repo

### PR DESCRIPTION
**Pull request by tarurar**

### CODE REVIEW

File: src/Lykke.Job.TransactionHandler/Utils/CashInOutOperationExt.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+using System.Linq;
+using Lykke.Job.TransactionHandler.Core.Contracts;
+using FeeType = Lykke.Service.OperationsRepository.AutorestClient.Models.FeeType;
+
+namespace Lykke.Job.TransactionHandler.Utils
+{
+    public static class CashInOutOperationExt
+    {
+        public static void AddFeeDataToOperation(this CashInOutOperation operation, CashInOutQueueMessage message,
+            ILog log)
```
Comment (MikhailGerasimov) [inline src/Lykke.Job.TransactionHandler/Utils/CashInOutOperationExt.cs]: Delete the log argument it is not used here

File: src/Lykke.Job.TransactionHandler.Core/Services/Fee/IFeeCalculationService.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Lykke.Job.TransactionHandler.Core.Services.Fee
+{
+    public interface IFeeCalculationService
+    {
+        Task<double> GetAmountNoFee(double initialAmount, string assetId, List<Contracts.Fee> fees);
```
Comment (MikhailGerasimov) [inline src/Lykke.Job.TransactionHandler.Core/Services/Fee/IFeeCalculationService.cs]: Add async suffix

File: src/Lykke.Job.TransactionHandler.Core/Contracts/LimitQueueMessage.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Generic;
+using JetBrains.Annotations;
+using Lykke.Job.TransactionHandler.Core.Domain.Exchange;
+using MessagePack;
+using Newtonsoft.Json;
+
+namespace Lykke.Job.TransactionHandler.Core.Contracts
```
Comment (MikhailGerasimov) [inline src/Lykke.Job.TransactionHandler.Core/Contracts/LimitQueueMessage.cs]: Add a description for types which will be shared. The name "contract" implies that it is a shared lib.

File: src/Lykke.Job.TransactionHandler.Core/Services/Fee/IFeeLogService.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using Lykke.Job.TransactionHandler.Core.Contracts;
+
+namespace Lykke.Job.TransactionHandler.Core.Services.Fee
+{
+    public interface IFeeLogService
```
Comment (MikhailGerasimov) [inline src/Lykke.Job.TransactionHandler.Core/Services/Fee/IFeeLogService.cs]: Add an async suffix

File: src/Lykke.Job.TransactionHandler.Services/Fee/FeeLogService.cs [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+            var tasks = feeDataSource.Trades.Select(x =>
+            {
+                var newItem = new FeeLogEntry
+                {
+                    OperationId = feeDataSource.Order.Id,
+                    Fee = x.Fees?.ToJson(),
+                    Type = FeeOperationType.LimitTrade
+                };
+
+                return _feelogRepository.CreateAsync(newItem);
```
Comment (MikhailGerasimov) [inline src/Lykke.Job.TransactionHandler.Services/Fee/FeeLogService.cs]: How many trades can be in the feeDataSource.Trades?  If there will be 100000 trades will this call DDoS azure table?

File: src/Lykke.Job.TransactionHandler.Services/ServiceExtensions.cs [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+using Lykke.Job.TransactionHandler.Core.Contracts;
+using Lykke.Job.TransactionHandler.Core.Services.Fee;
+
+namespace Lykke.Job.TransactionHandler.Services
+{
+    public static class ServiceExtensions
+    {
+        public static async Task<double> GetAmountNoFee(this IFeeCalculationService src, CashInOutQueueMessage message)
+        {
+            return await src.GetAmountNoFee(message.Amount.ParseAnyDouble(), message.AssetId, message.Fees);
```
Comment (MikhailGerasimov) [inline src/Lykke.Job.TransactionHandler.Services/ServiceExtensions.cs]: No need to do async await. Just return the result of the method call

File: src/Lykke.Job.TransactionHandler/ConvertExtensions.cs [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+using System.Linq;
+using Lykke.Job.TransactionHandler.Core.Contracts;
+using Lykke.Job.TransactionHandler.Core.Domain.Exchange;
+using Lykke.Job.TransactionHandler.Handlers;
+
+namespace Lykke.Job.TransactionHandler
+{
+    public static class ConvertExtensions
+    {
+        public static ClientTrade[] ToDomainOffchain(this LimitQueueItem.LimitOrderWithTrades item, string btcTransactionId, string clientId)
```
Comment (MikhailGerasimov) [inline src/Lykke.Job.TransactionHandler/ConvertExtensions.cs]: Can the result be IReadOnlyList?

File: src/Lykke.Job.TransactionHandler/Modules/JobModule.cs [hunk 8]
```diff
@@ -270,12 +274,8 @@ private void BindRepositories(ContainerBuilder builder)
...
 
             builder.RegisterInstance<IFeeLogRepository>(
-                new FeeLogRepository(AzureTableStorage<FeeLogEntity>.Create(
-                    _dbSettingsManager.ConnectionString(x => x.FeeLogsConnString), "OrdersFeeLog", _log)));
-
-            builder.RegisterInstance<ITransferLogRepository>(
-                new TransferLogRepository(
-                    AzureTableStorage<TransferLogEntity>.Create(_dbSettingsManager.ConnectionString(x => x.LogsConnString), "TransfersFeeLog", _log)));
+                new FeeLogRepository(AzureTableStorage<FeeLogEntryEntity>.Create(
+                    _dbSettingsManager.ConnectionString(x => x.FeeLogsConnString), "OperationsFeeLog", _log)));
```
Comment (MikhailGerasimov) [inline src/Lykke.Job.TransactionHandler/Modules/JobModule.cs]: Only the instance of AzureTableStorage should be registered with RegisterInstance method. FeeLogRepository can be registered with RegisterType<>

File: src/Lykke.Job.TransactionHandler.Core/Contracts/LimitQueueMessage.cs (hunk 3 above)
Comment (tarurar) [reply src/Lykke.Job.TransactionHandler.Core/Contracts/LimitQueueMessage.cs]: It is not a shared lib, at least now. It is rabbit message models which are being consumed from queues.

**Pull request merged**